### ISSUE TITLE
Programmers/Programmer’s/Programmers’

### ISSUE THREAD

**Issue by ke7ofi** [openscad/openscad.github.com#35]
The tagline is currently “The Programmers Solid 3D CAD Modeller”. I think that “The Programmer’s Solid 3D CAD Modeller” or “The Programmers’ Solid 3D CAD Modeller” would make more sense. (I prefer the first.)

Is there any particular reason for this? If not, I can make some changes and submit a PR.

(Also, the tagline seems a bit ungainly in general. Perhaps “Solid CAD for Programmers” or something else less redundant?)

**Comment by fredbar** [comment]
> Solid CAD for Programmers
> 
+1
3D seems unnecessary and that version is informative enough, with no frills.

**Comment by kintel** [comment]
I would prefer rewriting the tagline to keep up with the times.
OpenSCAD started out as a way of helping programmers create 3D designs, but we're currently reaching a bit farther than that.

**Comment by ke7ofi** [comment]
@kintel At that point, it’s more an introductory sentence (suitable for the first paragraph on the site) than a tagline.

**Comment by fredbar** [comment]
> Le 25 mai 2015 à 18:34, Marius Kintel <notifications@github.com> a écrit :
> 
> I would prefer rewriting the tagline to keep up with the times.
> OpenSCAD started out as a way of helping programmers create 3D designs, but we're currently reaching a bit farther than that.
> 
Then, maybe « programmable solid CAD » could fit?

**Comment by ke7ofi** [comment]
@fredbar That could just imply that it’s open-source, which is true, but not what makes it unique.

**Comment by kintel** [comment]
I tend to refer to OpenSCAD as the "HTML of 3D modelling" - perhaps some ideas in that direction would make sense.

**Comment by ke7ofi** [comment]
It’s a declarative language, so maybe “a declarative SCAD (or “solid CAD”) language”?

**Comment by fredbar** [comment]
Didn’t you mean « modeling »? ;)

**Comment by kintel** [comment]
Depends which flavour of English you're speaking :)

**Comment by ke7ofi** [comment]
Apparently “modelling” is the Commonwealth spelling and “modeling” is the American one, which is interesting considering that I’m from the US.

**Comment by fredbar** [comment]
Or the opposite. Anyway, that was a gentle pun at Marius, as the spelling is the reason that started this thread. No need to go anywhere with that. I would tend to totally ditch the word, if you ask me.

**Comment by ke7ofi** [comment]
So what do we want to do?

**Comment by ke7ofi** [comment]
@fredbar Looking at your account (and one commit), I get the joke. It’s interesting that we both use the opposite of our regionally expected spelling.

**Comment by ke7ofi** [comment]
Anyone interested in changing this? Should I submit a commit?

**Comment by kintel** [comment]
I'd still be more interested in ideas for new tag lines.

**Comment by ke7ofi** [comment]
I can't think of anything better than “Solid CAD for Programmers” (maybe “Coders” instead?), but it's probably at least better than the status quo.

**Comment by kintel** [comment]
We can ask the mailing list :)
Here's my 1 AM brain storming by itself:

3D Design Through Coding
design.code.create.
Programmable 3D markup
Descriptive Objects
3D design at your fingertips

**Comment by ke7ofi** [comment]
>3D Design Through Coding

sounds a bit weird

>design.code.create.

trendy-sounding, would trick browsers into going to that URL if someone searched for it

>Programmable 3D markup

I feel like OpenSCAD has more in common with HTML than Markdown etc.

>Descriptive Objects

This is definitely starting to sound like 01:00 copywriting.

>3D design at your fingertips

might imply touch interface, these days

**Comment by ke7ofi** [comment]
Looking back at this (I was slightly sleep-deprived as well.), I think “Solid CAD for Programmers” might sound exclusive, though that's not a major concern if there's something on the front page that explains how intuitive it is. Also, I bet there are a bunch of great puns regarding the fact that OpenSCAD is “object-oriented”.

**Comment by MichaelAtOz** [comment]
I like the "Solid CAD" descriptive, so:
Solid CAD for Coders' (SCC, SC4C)
(Apostrophe Rule 2a Regular nouns.  To show plural possession. - Had to look it up, English is my first & sometimes second language...)

Solid CAD Markup, 
maybe 
Solid CAD Markup Language 
(SCML or SCADML? I saw one reference of SCML to SystemC Modeling Library, " SystemC deliberately mimics the hardware description languages VHDL and Verilog, but is more aptly described as a system-level modeling language." - perhaps that is a good association?)

Is it just a language or is it the instantiated language or both?

Solid CAD Markup for Coders' (SCMC)
*Solid CAD - Markup for Coders' (SC-MC)

or contracted to 
CAD for Coders' (CAD4Coders, C4C)

*above is my fav.

**Comment by ke7ofi** [comment]
Most of these are directed at @MichaelAtOz:

Maybe your dialect is different from mine, but as far as I know, “Coders” wouldn't be possessive, as the “for” already took care of it. Which apostrophe configuration is “correct” in the current tagline depends on the intended meaning.

The current Wikipedia definition of a markup language is “a system for annotating a document in a way that is syntactically distinguishable from the text”, which applies to things like Markdown, but not (as far as I can tell) OpenSCAD. For that reason (the fact that OpenSCAD defines 3D objects instead of simply styling and structuring text), I'm hesitant to call it a markup language.

Are you trying to rename the program itself? I think OpenSCAD is fine, and a tagline doesn't necessarily have to define the title.

“CAD for Coders(')” inherits the problems of “Solid CAD for Programmers” without many of the benefits. Also, s/for/4/g has always annoyed me, though that's personal preference.

**Comment by t-paul** [comment]
From the 1AM list, I'd select the "design.code.create.". It did look nice on the flyers. It does not focus on a specific target audience. It's short enough and allows different arrangements to fit on stuff like t-shirts. It also seems a good base for extending the claim with more details. Finally it's not just a boring descriptive tag line ;-).

**Comment by ke7ofi** [comment]
It just seems a bit generic; I could list 50 projects that tagline would describe.

**Comment by MichaelAtOz** [comment]
@ke7ofi 

> Maybe your dialect is different from mine,...

As I said 'English is my ... sometimes second language...', and my worst subject. I was trying to cover the ambiguity of the title of this issue. I bow to the groups wisdom.

> annotating a document in a way that is syntactically distinguishable from the text

We are NOT marking up a text document, we are marking up geometry.
.html => web page; .scad => geometry

> Are you trying to rename the program itself?

No. A tag line/brand/catch-phrase.

> seems a bit generic

+1

**Comment by jordanbrown0** [comment]
Yes, I know this is nearly ten years old, but the web pager still has the unfortunate "Programmers" phraseology, where with that structure it should be possessive.

I have a knee-jerk negative reaction to "code" and words derived from it.  They connote secrecy and obscurity, which is the exact opposite of what we should be trying to do.

There's a fair minority doing 2D work these days, suggesting that dropping "3D" might be appropriate.  I'm not sure what "Solid" ever added to the mix, other than that it can be taken to imply lack of support for things like surface-only models, free-floating 1D and 2D in 3-space, et cetera.

I'd just say "CAD for Programmers".  (Which might be stretching the truth a little, since there's a lot of CAD-y things we can't do, but I don't think it's any worse than what we have now.)

**Comment by t-paul** [comment]
I think that's trying to fix a dead horse already buried in the desert for 10 years.

I'm not too fond of those 3 word slogans, but this still seems way better and working quite well (as I don't have anything against "code"). It's from a years old web site prototype that never happened...

![Screenshot From 2024-12-31 17-53-18](https://github.com/user-attachments/assets/7dbbaa68-2a56-49c4-a6c3-37e7f7fddbd4)
